src/resolve/resolved-dns-stub.c

```diff
@@ -386,6 +403,22 @@ static int on_dns_stub_packet(sd_event_source *s, int fd, uint32_t revents, void
         return 0;
 }
 
+static int set_dns_stub_common_socket_options(int fd) {
+        int r;
+
+        assert(fd >= 0);
+
+        r = setsockopt_int(fd, SOL_SOCKET, SO_REUSEADDR, true);
+        if (r < 0)
+                return r;
+
+        r = setsockopt_int(fd, IPPROTO_IP, IP_PKTINFO, true);
+        if (r < 0)
+                return r;
+
+        return setsockopt_int(fd, IPPROTO_IP, IP_RECVTTL, true);
```

Comment:
the latter two socktops are ipv4 specific, but socket might be ipv6. needs special casing to use the right socktops for the address family